### PR TITLE
GG-21320 Replicate ws/http(s) sessions across backends

### PR DESCRIPTION
**Pull request by nva**

### CODE REVIEW

File: modules/web-console/web-console-server/src/test/java/org/apache/ignite/console/web/socket/TransitionServiceSelfTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    /** Browsers service. */
+    @MockBean
+    private BrowsersService browsersSrvc;
+
+    @Captor
+    private ArgumentCaptor<WebSocketEvent<Announcement>> annCaptor;
+
+    /**
+     */
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/test/java/org/apache/ignite/console/web/socket/TransitionServiceSelfTest.java]: replace with `/** */`

File: modules/web-console/web-console-server/src/test/java/org/apache/ignite/console/MockConfiguration.java [hunk 2]
```diff
@@ -83,6 +105,10 @@ public Ignite igniteInstance() {
             @Override public IgniteTransactions transactions() {
                 return txs;
             }
+
+            @Override public IgniteCluster cluster() {
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/test/java/org/apache/ignite/console/MockConfiguration.java]: Missing  /** inherited javadocs */

File: modules/web-console/web-console-server/src/test/java/org/apache/ignite/console/MockConfiguration.java [hunk 3]
```diff
@@ -83,6 +105,10 @@ public Ignite igniteInstance() {
             @Override public IgniteTransactions transactions() {
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/test/java/org/apache/ignite/console/MockConfiguration.java]: Missing /** inherited javadocs */

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/TransitionService.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+import org.apache.ignite.messaging.MessagingListenActor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import static org.apache.ignite.console.websocket.WebSocketEvents.ADMIN_ANNOUNCEMENT;
+import static org.apache.ignite.events.EventType.EVTS_DISCOVERY;
+
+/**
+ *
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/TransitionService.java]: Missing javadocs

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java [hunk 5]
```diff
@@ -0,0 +1,531 @@
...
+
+    /**
+     * @param accIds Account ids.
+     * @param demo is demo stats.
+     */
+    private void sendAgentStats(Set<UUID> accIds, boolean demo) {
+        for (UUID accId : accIds) {
+            UserKey key = new UserKey(accId, demo);
+            WebSocketResponse stats = collectAgentStats(key);
+
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java]: remove not needed new line

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java [hunk 6]
```diff
@@ -0,0 +1,531 @@
...
+     * Send request to locally connected agent.
+     * @param req Request.
+     * @throws IllegalStateException If connected agent not founded.
+     */
+    void sendLocally(AgentRequest req) throws IllegalStateException, IOException {
+        WebSocketSession ses = findLocalAgent(req.getKey()).orElseThrow(IllegalStateException::new);
+
+        WebSocketEvent evt = req.getEvent();
+
+        log.debug("Found local agent session [session=" + ses + ", event=" + evt + "]");
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentsService.java]: missing `if (log.isDebugEnabled())`

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentKey.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+     */
+    public String getClusterId() {
+        return clusterId;
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/AgentKey.java]: add new line before if

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/TransitionService.java [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+     */
+    public TransitionService(Ignite ignite, TransactionManager txMgr, AgentsService agentsSrvc, BrowsersService browsersSrvc) {
+        this.ignite = ignite;
+        this.txMgr = txMgr;
+        
+        this.agentsSrvc = agentsSrvc;
+        this.browsersSrvc = browsersSrvc;
+
+        this.txMgr.registerStarter(() -> {
+            backendByAgent = new OneToManyIndex<>(ignite, "wc_backends");
```
Comment (smoldachev) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/TransitionService.java]: Another reason to move the code to the repository.

File: modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/TransitionService.java [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+    /**
+     * @param req Request.
+     * @param prefix Prefix.
+     * @param e Exception.
+     */
+    private void responseWithError(AgentRequest req, String prefix, Throwable e) {
+        try {
+            ignite.message(ignite.cluster().forNodeId(req.getSrcNid()))
+                .send(SEND_RESPONSE, req.getEvent().withError(prefix, e));
+        } catch (Exception e1) {
```
Comment (akuznetsov-gridgain) [inline modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/TransitionService.java]: `catch()` should be on new line

**Pull request merged**